src/core/animation-loop.js

```diff
@@ -122,171 +115,140 @@ export default class AnimationLoop {
   stop() {
     // console.debug(`Stopping ${this.constructor.name}`);
     if (this._animationFrameId) {
-      this._finalizeContext();
+      this._finalizeCallbackData();
       cancelAnimationFrame(this._animationFrameId);
       this._animationFrameId = null;
     }
     return this;
   }
 
   // Resize canvas in "CSS coordinates" (may be different from device coords)
-  // NOTE: No effect on headless contexts
   // @param {Number} width, height - new width and height of canvas in CSS coordinates
   resizeCanvas(width, height) {
-    this._resizeCanvas(width, height);
+    resizeCanvas({width, height});
```

Comment:
> why this function is calling itself recursively

@shaojingli Got an email with your question but can't see it in github. The answer is that it is actually calling a global function with the same name (since there is no `this.`). 

Agree it is a bit too subtle, the alternative is to give either the global function or the method a less natural name.